src/Simulation/DepletionVoltage.jl

```diff
@@ -0,0 +1,164 @@
+# """
+#     _adapt_weighting_potential_to_electric_potential_grid!(sim::Simulation, contact_id::Int)
+# 
+# Interpolates the [`WeightingPotetial`](@ref) of the [`Contact`](@ref) with id `contact_id`
+# onto the [`Grid`](@ref) of the [`ElectricPotential`](@ref) and updates it until it converges.
+# 
+# ## Arguments 
+# * `sim::Simulation{T}`: [`Simulation`](@ref) for which the [`WeightingPotential`] should be adapted.
+# * `contact_id::Int`: The `id` of the [`Contact`](@ref) at which the [`WeightingPotential`] should be adapted.
+# """
+function _adapt_weighting_potential_to_electric_potential_grid!(sim::Simulation, contact_id::Int)
+    @assert !ismissing(sim.electric_potential) "Electric potential missing"
+    if ismissing(sim.weighting_potentials[contact_id]) calculate_weighting_potential!(sim, contact_id, verbose = false) end
+    if sim.weighting_potentials[contact_id].grid != sim.electric_potential.grid
+        sim.weighting_potentials[contact_id] = sim.weighting_potentials[contact_id][sim.electric_potential.grid]
+        #apply_initial_state!(sim, WeightingPotential, contact_id, sim.electric_potential.grid)
+        update_till_convergence!(sim, WeightingPotential, contact_id)
+    end
+end
+
+"""
+    get_depletion_voltage( sim::Simulation{T}, contact_id::Int, potential_range::AbstractRange; kwargs... )::T
+
+Returns the potential (in V) needed at the [`Contact`](@ref) with id `contact_id`
+to fully deplete the detector in a given [`Simulation`](@ref). For this, all other
+contact potentials are set to `0` and the potential at the specified contact is
+increased or decreased according to the `potential_range`. The depletion voltage
+is set to the potential for which a previously undepleted detector becomes depleted,
+or for which a previously depleted detector becomes undepleted.
+
+## Arguments 
+* `sim::Simulation{T}`: [`Simulation`](@ref) for which the depletion voltage should be determined.
+* `contact_id::Int`: The `id` of the [`Contact`](@ref) at which the potential is applied.
+* `potential_range::AbstractRange`: Range of potentials to be tested.
+    
+## Keywords
+* `verbose::Bool = true`: Activate or deactivate additional info output. Default is `true`.
+
+## Example 
+```julia
+using SolidStateDetectors
+sim = Simulation(SSD_examples[:InvertedCoax])
+calculate_electric_potential!(sim)
+get_depletion_voltage(sim, 2, 1600:1:2500)
+```
+
+!!! warn
+    This method only works if the initial `sim` was calculated for a case in which the detector is fully depleted.
+
+!!! note
+    The accuracy of the result depends on the precision of the initial simulation.
+    
+See also [`is_depleted`](@ref).
+"""
+function get_depletion_voltage(sim::Simulation{T}, contact_id::Int,
+            potential_range::AbstractRange = range(extrema(broadcast(c -> c.potential, sim.detector.contacts))..., length = 1001);
```

Comment:
The problem with `potential_range` is the sign.
We have seen this in the case of the inverted coaxial point contact detector: you can achieve depletion by applying a negative potential of around -2000V to the core, or by applying a positive potential of around +2000V to the mantle.

In the config file, the potential applied to the core is 0V, whereas the potential applied to the mantle is +3500V.
Thus, `potential_range` is ranging from 0 to +3500V. However, when we look for the potential that we have to apply to the **CORE** to deplete the detector, we would expect the result to be negative and, thus, not in the range from 0 to +3500V. 

It might make sense to ensure that the user do not keep on increasing the range to unreasonably high voltages for cases in which the detector is not operated in reversed bias voltage. 

Any thought on how to deal with this?